app/src/main/cpp/Controller.h

```diff
@@ -52,6 +53,8 @@ struct Controller {
   float immersiveAxes[kControllerMaxAxes];
   uint32_t numAxes;
   uint32_t numHaptics;
+  mozilla::gfx::VRControllerType type;
+  mozilla::gfx::TargetRayMode targetRayMode;
```

Comment:
`moz_external_vr.h` and `mozilla::gfx` types should not leak outside of ExternalVR.cpp

Reply:
I am going to mirror these types in Device.h